### ISSUE TITLE
4.x: Merging issues with config

### ISSUE THREAD

**Issue by dereuromark** [cakephp/cakephp#13224]
This is a (multiple allowed):

* [x] bug
* [ ] enhancement
* [ ] feature-discussion (RFC)

* CakePHP Version: 4.x dev

### What you did
I have a plugin from 3.x, upgraded to 4.x that has this in tests bootstrap:
```php
ConnectionManager::setConfig('test', [
	...
	'database' => getenv('db_database') ?: null,
```

both null and false worked fine so far to let the internal defaults take over:

```php
    protected $_baseConfig = [
        'persistent' => false,
        'username' => null,
        'password' => null,
        'database' => ':memory:',
```

### What happened
But now in 4.x when trying to run tests:

> PHP Fatal error:  Uncaught TypeError: file_exists() expects parameter 1 to be a valid path, null given in /.../vendor/cakephp/cakephp/src/Database/Driver/Sqlite.php:69

### What you expected to happen
It should use the core defaults here.

### Also

putenv('db_dsn=sqlite::memory:');

is in there, which then should take over completely, this is not happening either.

**Comment by dereuromark** [comment]
It looks like public static function parseDsn(?string $config = null): array
is not invoked anymore in 4.x here.

**Comment by markstory** [comment]
Are you putting a dsn into the `database` key?

**Comment by markstory** [comment]
Fixed in #13242
